sdk/core/core-client-rest/src/urlHelpers.ts

```diff
@@ -57,3 +59,26 @@ export function buildRequestUrl(
       .replace(/([^:]\/)\/+/g, "$1")
   );
 }
+
+export function buildBaseUrl(baseUrl: string, options: RequestParameters): string {
+  if (!options.pathParameters) {
+    return baseUrl;
+  }
+  const pathParams = options.pathParameters;
+  for (const key of Object.keys(pathParams)) {
+    const param = pathParams[key] as any;
+    if (param === undefined || param === null) {
+      throw new Error(`Query parameters ${key} must not be undefined or null`);
```

Comment:
This are Path parameters not Query, please update the message. Same in line 74.

I think throwing here is a good idea, since replacing a `{variable}` template with null or undefined will give us an invalid path. In query string parameters we can just skip them since it won't end up as an invalid/malformed url